bpf/lib/lb.h

```diff
@@ -644,7 +645,16 @@ struct lb6_service *lb6_lookup_service(struct lb6_key *key,
 		if (!scope_switch || !lb6_svc_is_two_scopes(svc))
 			return svc;
 		key->scope = LB_LOOKUP_SCOPE_INT;
+		key->proto = orig_proto;
 		svc = map_lookup_elem(&LB6_SERVICES_MAP_V2, key);
+
+#if defined(ENABLE_SERVICE_PROTOCOL_DIFFERENTIATION)
+		/* Also check for ANY protocol for internal scope lookups */
+		if (!svc && key->proto != 0) {
+			key->proto = 0;
+			svc = map_lookup_elem(&cilium_lb6_services_v2, key);
```

Comment:
Just for consistency, I think on v1.17 this should be

```suggestion
			svc = map_lookup_elem(&LB6_SERVICES_MAP_V2, key);
```

(same below for the IPv4 path)

Reply:
Oops, I was missing that. Thanks!